File: SunnyFlamingo/ClientApp/src/app/models/goods/computerTechnologies/computerParts/videoCardModel.ts
import { ComputerPartModel } from "../computerPartModel";

export class VideoCardModel extends ComputerPartModel{
  videoSize: number;
  driveInterface: string;
  videoMemoryCapacity: number;
  constructor(options: {
    name: string;
    price: number;
    isAvailable: string;
    description: string;
    producerId: string;
    manufacturerId: string;
    material: string;
    color: string;
    imageIds: string[];
    videoSize: number;
    driveInterface: string;
    videoMemoryCapacity: number;
  }){
    super(options);
    this.videoSize = +options.videoSize;
    this.driveInterface = options.driveInterface;
    this.videoMemoryCapacity = +options.videoMemoryCapacity;
  }
}

File: SunnyFlamingo/Models/Goods/ComputerTechnologies/ComputerParts/VideoCardModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SunnyFlamingo.Models
{
    public class VideoCardModel: ComputerPartModel
    {
        public int VideoSize { get; set; }
        public string DriveInterface { get; set; }
        public int VideoMemoryCapacity { get; set; }
    }
}
